unlock-app/src/hooks/useMarkAsCheckedIn.js

```diff
@@ -0,0 +1,41 @@
+import { useDispatch } from 'react-redux'
+
+import { useState } from 'react'
+import { waitForWallet, dismissWalletCheck } from '../actions/fullScreenModals'
+
+/**
+ * This hooks marks a key as checked-in
+ * By saving a check in data as metadata
+ * @param {*} address
+ */
+export const useMarkAsCheckedIn = (walletService, config, key) => {
+  const dispatch = useDispatch()
+  const [checkedIn, setCheckedIn] = useState(false)
+  const [error, setError] = useState(false)
+
+  const useMarkAsCheckedIn = () => {
```

Comment:
```suggestion
  const markAsCheckedIn = () => {
```

I think the function the hook returns should be named differently from the hook itself